### PR TITLE
Mdrestext BDRE-147

### PR DESCRIPTION
**Pull request by bdrekapil**

### CODE REVIEW

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/CrawlerAPI.java [hunk 1]
```diff
@@ -46,16 +46,15 @@
...
     private ProcessDAO processDAO;
-    @Autowired
-    private PropertiesDAO propertiesDAO;
 
     @RequestMapping(value = {"/", ""}, method = RequestMethod.POST)
 
-    public
     @ResponseBody
-    RestWrapper createCrawlerProcess(@ModelAttribute("crawlerInfo")
+    public RestWrapper createCrawlerProcess(@ModelAttribute("crawlerInfo")
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/CrawlerAPI.java:57]: MAJOR The Cyclomatic Complexity of this method "createCrawlerProcess" is 13 which is greater than 10 authorized. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FlumePropertiesAPI.java [hunk 2]
```diff
@@ -148,8 +146,10 @@ RestWrapper insert(@ModelAttribute("generalconfig")
             restWrapper = new RestWrapper(generalConfig, RestWrapper.OK);
             LOGGER.info("Record with configGroup:" + generalConfig.getConfigGroup() + " inserted in Jtable by User:" + principal.getName() + generalConfig);
         } catch (Exception e) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FlumePropertiesAPI.java:148]: CRITICAL Either log or rethrow this exception. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1166)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FlumePropertiesAPI.java [hunk 3]
```diff
@@ -43,15 +43,14 @@
...
     private ProcessDAO processDAO;
-    @Autowired
-    private PropertiesDAO propertiesDAO;
 
     @RequestMapping(value = {"/createjobs"}, method = RequestMethod.POST)
 
-    public
-    @ResponseBody
+    @ResponseBody public
     RestWrapper createJob(@RequestParam Map<String, String> map, Principal principal) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FlumePropertiesAPI.java:54]: MAJOR The Cyclomatic Complexity of this method "createJob" is 11 which is greater than 10 authorized. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Remove the unused method parameter(s) "principal". [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java [hunk 4]
```diff
@@ -46,15 +46,14 @@
...
     @Autowired
     private ProcessDAO processDAO;
 
     @RequestMapping(value = {"/createjobs"}, method = RequestMethod.POST)
 
-    public
-    @ResponseBody
+
+    @ResponseBody public
     RestWrapper createJobs(@RequestParam Map<String, String> map, Principal principal) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java:57]: MAJOR The Cyclomatic Complexity of this method "createJobs" is 18 which is greater than 10 authorized. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Remove the unused method parameter(s) "principal". [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java [hunk 5]
```diff
@@ -204,6 +201,7 @@ RestWrapper insert(@ModelAttribute("generalconfig")
             restWrapper = new RestWrapper(generalConfig, RestWrapper.OK);
             LOGGER.info("Record with configGroup:" + generalConfig.getConfigGroup() + " inserted in Jtable by User:" + principal.getName() + generalConfig);
         } catch (Exception e) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java:203]: CRITICAL Either log or rethrow this exception. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1166)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java [hunk 6]
```diff
@@ -89,35 +87,36 @@ RestWrapper createJobs(@RequestParam Map<String, String> map, Principal principa
             String key = string.substring(splitIndex + 1, string.length());
             LOGGER.debug("key is " + key);
 
-            if (string.startsWith("type_genArg")) {
-//                String keySplit[] = key.split(".");
-//                LOGGER.debug("keySplit length"+keySplit.length);
+            if (string.startsWith("type_genArg") && map.get(string).split(",").length == 3) {
                 fieldTypeCounter = Integer.parseInt(string.substring((string.lastIndexOf(".") + 1), string.length()));
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java:91]: MAJOR Remove those useless parentheses. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java [hunk 7]
```diff
@@ -89,35 +87,36 @@ RestWrapper createJobs(@RequestParam Map<String, String> map, Principal principa
...
                 }
+                unifiedDate.append(date.getTime() + "," + date2.getTime() + "," + dateContent[2]);
+                jpaProperties = Dao2TableUtil.buildJPAProperties("data", "args." + fieldArgCounter, unifiedDate.toString(), "Generated Argument");
+                childProps.add(jpaProperties);
+                jpaProperties = Dao2TableUtil.buildJPAProperties("data", "data-gen-id." + fieldArgCounter, map.get("type_generatedType." + fieldTypeCounter), "Generated Type");
+                childProps.add(jpaProperties);
+                fieldArgCounter++;
+            }
+            else if(string.startsWith("type_genArg")){
+                fieldTypeCounter = Integer.parseInt(string.substring((string.lastIndexOf(".") + 1), string.length()));
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java:110]: MAJOR Remove those useless parentheses. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataLoadAPI.java [hunk 8]
```diff
@@ -89,8 +94,7 @@ RestWrapper list(Principal principal) {
 
     @RequestMapping(value = {"/createjobs"}, method = RequestMethod.POST)
 
-    public
-    @ResponseBody
+    @ResponseBody public
     RestWrapper createJob(@RequestParam Map<String, String> map, Principal principal) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataLoadAPI.java:98]: MAJOR The Cyclomatic Complexity of this method "createJob" is 35 which is greater than 10 authorized. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by bdre-bda** [comment]
SonarQube analysis reported 16 issues:
- CRITICAL 2 critical
- MAJOR 10 major
- MINOR 4 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
- MINOR [Remove this unused import 'com.wipro.ats.bdre.md.dao.PropertiesDAO'.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/CrawlerAPI.java#L21) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessImportCheck)
- MINOR [Remove this unused import 'com.wipro.ats.bdre.md.dao.PropertiesDAO'.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataGenAPI.java#L21) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessImportCheck)
- MAJOR [The Cyclomatic Complexity of this method "buildTablesFromMap" is 12 which is greater than 10 authorized.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataImportAPI.java#L289) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
- MAJOR [Reduce the total number of break and continue statements in this loop to use at most one.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/DataLoadAPI.java#L117) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS135)
- MINOR [Remove this unused import 'com.wipro.ats.bdre.md.dao.PropertiesDAO'.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FileMoniterAPI.java#L19) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessImportCheck)
- MINOR [Remove this unused import 'com.wipro.ats.bdre.md.dao.PropertiesDAO'.](https://github.com/WiproOpenSourcePractice/openbdre/blob/04f3ed1d333967aa913b3bd4b62eec67aedb88b5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/FlumePropertiesAPI.java#L21) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by wiprodev** [comment]
@bdrekapil If possible please do this in your local machine before opening pull request

```
mvn sonar:sonar -s settings.xml -P hdp22 org.jacoco:jacoco-maven-plugin:prepare-agent -DskipTests=true -Dsonar.language=java  -Dsonar.analysis.
mode=incremental -Dsonar.host.url=https://sonar.openbdre.org -Dsonar.login=<user> -Dsonar.password=<password> -Dsonar.issuesReport.html.enable=true
```

**Comment by bdrekapil** [comment]
I tried that before submitting pull request, but didn't work.

**Pull request closed without merging**